src/Core/Index/IndexMap.inl

```diff
@@ -116,7 +145,7 @@ inline bool IndexMap< T >::remove( const uint i ) {
 /// ACCESS
 /// ===============================================================================
 template < typename T >
-inline T IndexMap< T >::at( const Index& idx ) const {
+inline const T& IndexMap< T >::at( const Index& idx ) const {
     typename std::deque< Index >::const_iterator it = std::find( m_index.begin(), m_index.end(), idx );
     CORE_ASSERT( it != m_index.end(), "Index not found" );
     return m_data.at( it - m_index.begin() );

```

Comment:
Would be cleaner to use std::distance here, no ?
http://en.cppreference.com/w/cpp/iterator/distance